Bug 1312230 - fix image import command to reflect the actual status

soltysh opened the pull request:
Currently we *always* return `The import completed successfully` no matter what actually happened. This PR small check to modify slightly this information to better reflect the actual status.

Fixes [bug 1312230](https://bugzilla.redhat.com/show_bug.cgi?id=1312230).

@openshift/cli-review ptal

juanvallejo commented on pkg/cmd/cli/cmd/importimage.go:
```diff
@@ -147,7 +148,11 @@ func (o *ImportImageOptions) Run() error {
 	default:
-		fmt.Fprint(o.out, "The import completed successfully.\n\n")
+		if wasError(result) {
+			fmt.Fprint(o.out, "The import did not complete successfully, please check below for details.\n\n")
```
Hm, we should be outputting this through stderr

soltysh commented:
Comments addressed.

smarterclayton commented on pkg/cmd/cli/cmd/importimage.go:
```diff
@@ -147,7 +150,11 @@ func (o *ImportImageOptions) Run() error {
 	default:
-		fmt.Fprint(o.out, "The import completed successfully.\n\n")
+		if wasError(result) {
+			fmt.Fprint(o.errout, "The import did not complete successfully, please check below for details.\n\n")
```
Don't say "successfully" if there was an error.   Also, we try to exit with "error: ", so I don't think this line would be correct.  It would be better to actually print out the error condition directly to the user water than print the describe.

soltysh commented:
@smarterclayton updated, current output:
```
$ oc import-image pyfake 
The import completed with following errors:
- latest: you may not have access to the Docker image "wsun/non-existen-image".

Name:			pyfake
Namespace:		test
...
```

soltysh commented:
Rebased @smarterclayton || @openshift/cli-review ptal

fabianofranz commented:
LGTM

soltysh commented:
Merging based on previous approval 

[merge]

mfojtik commented:
LGTM, good stuff @soltysh

soltysh commented:
> LGTM, good stuff @soltysh

I'm just fixing bugs :wink:

smarterclayton commented:
Errors should print with an "error: " prefix (and print last) if you encounter them.

smarterclayton commented:
Removing merge, it doesn't quite explain that errors occurred in a way that jumps out at the user.

smarterclayton commented:
Basically from the output it's not obvious that the action actually failed.  Either:

one error:

```
error: Import of tag failed: <error>
```

multiple errors:

```
error: Import of tag failed:
* ...
* ...
```

or errors plus output:

```
Import partially successful:

Describe output:

error: Import of tag X failed ...
```

soltysh commented:
@smarterclayton updated, I dropped printing errors, since we already do that next to every tag (sic!), so I've just changed the successful message to one with information there was errors. Here's the current output: 
```
$ oc import-image pyfake 
The import completed with errors.

Name:			pyfake
Namespace:		test
Created:		7 minutes ago
Labels:			<none>
Annotations:		openshift.io/image.dockerRepositoryCheck=2016-09-29T10:40:23Z
Docker Pull Spec:	172.30.54.29:5000/test/pyfake
Unique Images:		0
Tags:			2

latest
  tagged from wsun/non-existen-image

  ! error: Import failed (Unauthorized): you may not have access to the Docker image "wsun/non-existen-image"
      7 minutes ago

3.3
  tagged from wsun/pythonlol-33-centos

  wsun tests

  ! error: Import failed (Unauthorized): you may not have access to the Docker image "wsun/pythonlol-33-centos"
      7 minutes ago
```
Is it more to your liking?

smarterclayton commented:
Yes that's excellent.

On Thu, Sep 29, 2016 at 8:12 AM, OpenShift Bot <notifications@github.com>
wrote:

> continuous-integration/openshift-jenkins/test SUCCESS (
> https://ci.openshift.redhat.com/jenkins/job/test_pr_origin/9477/)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/10847#issuecomment-250448989>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_p7vv2eVkjkljEmveES3Yz8KuRTNeks5qu6swgaJpZM4J36t1>
> .
>

soltysh commented:
[merge] 

upon approval
